packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -1532,15 +1532,31 @@ export class BaseResolversVisitor<
     });
     const typeName = node.name as any as string;
     const parentType = this.getParentTypeToUse(typeName);
-    const isRootType = [
-      this.schema.getQueryType()?.name,
-      this.schema.getMutationType()?.name,
-      this.schema.getSubscriptionType()?.name,
-    ].includes(typeName);
 
-    const fieldsContent = node.fields.map((f: any) => f(node.name));
+    const rootType = ((): false | 'query' | 'mutation' | 'subscription' => {
+      if (this.schema.getQueryType()?.name === typeName) {
+        return 'query';
+      }
+      if (this.schema.getMutationType()?.name === typeName) {
+        return 'mutation';
+      }
+      if (this.schema.getSubscriptionType()?.name === typeName) {
+        return 'subscription';
+      }
+      return false;
+    })();
+
+    const fieldsContent = (node.fields as unknown as FieldDefinitionPrintFn[]).map(f => {
```

Comment:
I have to inline `node.fields as unknown as FieldDefinitionPrintFn[]` here because it's the easiest way to communicate where this function comes from. If there's a better way, let me know!